sql/core/src/test/scala/org/apache/spark/sql/internal/CatalogSuite.scala

```diff
@@ -1050,6 +1052,16 @@ class CatalogSuite extends SharedSparkSession with AnalysisTest with BeforeAndAf
       func2.className.startsWith("org.apache.spark.sql.internal.CatalogSuite"))
   }
 
+  test("SPARK-46145: listTables does not throw exception when the table or view is not found") {
+    val impl = spark.catalog.asInstanceOf[CatalogImpl]
+    for (temp <- Seq(true, false)) {
+      val row =
+        new GenericInternalRow(Array(UTF8String.fromString("db"), UTF8String.fromString("t"), temp))
```

Comment:
shall we make the name more explicit? `non_existing`

Reply:
and let's set schema to empty string for temp views.

Reply:
done